src/FacebookPlayer.js

```diff
@@ -160,11 +168,11 @@ class FacebookPlayer extends React.Component {
     playerDiv.classList.add('fb-video');
     playerDiv.id = playerId;
     playerDiv.setAttribute('data-href', 'https://www.facebook.com/facebook/videos/' + videoId);
-    playerDiv.setAttribute('data-allowfullscreen', allowfullscreen || 'false');
-    playerDiv.setAttribute('data-autoplay', autoplay || 'false');
-    playerDiv.setAttribute('data-width', width || 'auto');
-    playerDiv.setAttribute('data-show-text', showText || 'false');
-    playerDiv.setAttribute('data-show-captions', showCaptions || 'false');
+    playerDiv.setAttribute('data-allowfullscreen', allowfullscreen);
+    playerDiv.setAttribute('data-autoplay', autoplay);
+    playerDiv.setAttribute('data-width', width);
```

Comment:
If I understand your question, this happens on lines 24-30. Default props are being used by React if they are not passed into the component.

https://facebook.github.io/react/docs/typechecking-with-proptypes.html#default-prop-values